src/layer/poi.js

```diff
@@ -53,6 +53,14 @@ var iconDefs = {
     color: Color.poi.infrastructure,
     description: "School",
   },
+  townhall: {
+    classes: {
+      town_hall: ["townhall"],
+    },
+    sprite: "poi_town_hall",
+    color: Color.poi.infrastructure,
+    description: "Town hall",
```

Comment:
As far as I can tell, American maps tend to label this as “city hall” rather than “town hall”, maybe because formally, every state has cities but only some have towns (while others have villages etc.). Or we could split the difference with “City or town hall”.

There’s still the minor issue of `amenity=townhall` being used for [county headquarters](https://taginfo.openstreetmap.org/tags/townhall%3Atype=county#overview) that aren’t courthouses, but that won’t be as much of a problem once we use the same icon for government buildings and rename this entry to “Government office” or somesuch.

Reply:
I'd be fine with "City hall" or "City or town hall" in the legend. I went with town only because that's closer to the OSM-ism for it.

I don't see other uses of the tag as too much of a blocker. The defined use seems predominant, and I view a lot of those oddball ones as coming from OSM's lack of an `amenity` tag for government centers besides `amenity=townhall`. Heck, even the state capitol of California is tagged `amenity=townhall`: https://www.openstreetmap.org/relation/20084. The `townhall:type=county` only has 84 uses in the database anyways, compared to ~150k `amenity=townhall` uses total, and many of those are outside the US, where the definition of a county can be different.

Reply:
I would prefer something like city/town hall rather than something more technical like "municipal administration building" :grimacing: 

I would suggest that there are far more town halls than city halls in existence!

Reply:
Unscientific overpass queries: 2687 `amenity=townhall` in the US contain "Town Hall" in the name ([link](https://overpass-turbo.eu/s/1tqB)), 4284 `amenity=townhall` in the US contain "City Hall" in the name ([link](https://overpass-turbo.eu/s/1tqC)). Seems to come from many states having few "towns", so even places with small populations are still technically "cities". The "town halls" appear concentrated in a few states, mostly in the Northeast but also noticeably SC, WI, and MD, that have lots of "towns", whereas "City Halls" are more evenly distributed (even places with lots of towns have some cities). This seems to support naming the icon "City hall" or maybe "City/town hall"

Reply:
Good evidence, I stand corrected.

Reply:
> I would prefer something like city/town hall rather than something more technical like "municipal administration building" 😬

Oh yeah, as long as this entry corresponds to `amenity=townhall`, it should be some kind of “hall”. But I think it’s only a matter of time before we reuse this icon for `office=government` (which includes state capitols), at which point we’d generalize this legend entry to correspond to either `amenity=townhall` or `office=government` (“City hall or other government office”).

Reply:
> But I think it’s only a matter of time before we reuse this icon for office=government (which includes state capitols), at which point we’d generalize this legend entry to correspond to either amenity=townhall or office=government (“City hall or other government office”).

That's actually a really good idea, and is aligned with our concept for simpler icons.  We could keep the town/city halls at lower zooms than the other government offices might show up at.

Reply:
I switched the description to "City hall". Figured more concise was better rather than going with "City or town hall". I think people will understand it applies to other municipality types.